otcextensions/sdk/cce/v3/node_pool.py

```diff
@@ -0,0 +1,223 @@
+    # Enable or disable auto scaling
+    enable = resource.Body('enable', type=bool)
+    # Minimum number of nodes after a scale-up if auto-scaling is enabled.
+    min_node_count = resource.Body('minNodeCount', type=int)
+    # Maximum number of nodes after scale-up, if auto-scaling is enabled.
+    # The value must be equal or greater than min_node_count and cannot
+    # exceed the maximum number of nodes of the cluster.
+    max_node_count = resource.Body('maxNodeCount', type=int)
+    # Node pool weight. A higher weight indicates a higher priority
+    # in scale-up.
+    priority = resource.Body('priority', type=int)
+    # Interval in minutes during which nodes added aafter a scale-up
+    # will not be deleted.
+    scale_down_cooldown_time = resource.Body('scaleDownCooldownTime', type=int)
+
+
+class NodeManagementSpec(resource.Resource):
+    # ECS groupt id of the ECS group to which those nodes belong
+    # after creation.
+    ecs_group_id = resource.Body('serverGroupReference')
+
+
+class SubnetIdSpec(resource.Resource):
+    # ID of the subnet to which the NIC belongs
+    subnet_id = resource.Body('subnetId')
+
+
+class NodeNicSpec(resource.Resource):
+    # Description about the primary Nic
+    primary_nic = resource.Body('primaryNic', type=SubnetIdSpec)
+
+
+class NodeTemplateSpec(resource.Resource):
+    # Properties
+    # Name of the AZ where the node resides.
+    # If availability_zone is set to: random, nodes can be created in any AZ
+    availability_zone = resource.Body('az')
+    # Billing mode of a node. Currently, only pay-per-use is supported.
+    billing_mode = resource.Body('billingMode', type=int, default=0)
+    # Number of nodes.
+    count = resource.Body('count', type=int)
+    # Data disk parameters of a node. At present, only one data
+    # disk can be configured
+    data_volumes = resource.Body('dataVolumes', type=list,
+                                 list_type=DataVolumeSpec)
+    # ID of the Dedicated Host to which nodes will be scheduled
+    dedicated_host = resource.Body('dedicatedHostId')
+    # ID of the ECS group where the CCE node can belong to
+    ecs_group = resource.Body('ecsGroupId')
+    # Extended parameters in key-value format
+    extend_params = resource.Body('extendParam', type=ExtendParamSpec)
+    # The node is created in the specified fault domain.
+    fault_domain = resource.Body('faultDomain')
+    # Flavor (mandatory)
+    flavor = resource.Body('flavor')
+    # Elastic IP address parameters of a node.
+    floating_ip = resource.Body('publicIP', type=PublicIPSpec)
+    # Kubernetes tags
+    k8s_tags = resource.Body('k8sTags', type=dict)
+    # Parameters for logging in to the node.
+    login = resource.Body('login')
+    node_nic_spec = resource.Body('nodeNicSpec', type=NodeNicSpec)
+    # Boolean: if node is offloading all its components
+    offload_node = resource.Body('offloadNode', type=bool)
+    # Operating System of the node. EulerOS and CentOS are supported.
```

Comment:
CentOs is supported only from k8s 1.17+
we should warn about this somewhere

Reply:
added